tst/standard/attr.tst

```diff
@@ -1509,6 +1509,110 @@ gap> ChromaticNumber(a);
+4
+gap> gr := Digraph([[2, 4, 7, 3], [3, 5, 8, 1], [1, 6, 9, 2],
+> [5, 7, 1, 6], [6, 8, 2, 4], [4, 9, 3, 5], [8, 1, 4, 9], [9, 2, 5, 7],
+> [7, 3, 6, 8]]);;
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmLawler);
+3
+gap> gr := DigraphSymmetricClosure(ChainDigraph(5));
+<immutable symmetric digraph with 5 vertices, 8 edges>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmLawler);
+2
+gap> gr := DigraphFromGraph6String("KmKk~K??G@_@");
+<immutable digraph with 12 vertices, 42 edges>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmLawler);
+4
+gap> gr := CycleDigraph(7);
+<immutable cycle digraph with 7 vertices>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmLawler);
+3
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmLawler);
+3
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmLawler);
+3
+
+#  Test ChromaticNumber Byskov 
+gap> ChromaticNumber(Digraph([[1]]), DigraphColouringAlgorithmByskov);
+Error, the argument <D> must be a digraph with no loops,
+gap> ChromaticNumber(NullDigraph(10), DigraphColouringAlgorithmByskov);
+1
+gap> ChromaticNumber(CompleteDigraph(10), DigraphColouringAlgorithmByskov);
+10
+gap> ChromaticNumber(CompleteBipartiteDigraph(5, 5), DigraphColouringAlgorithmByskov);
+2
+gap> ChromaticNumber(DigraphRemoveEdge(CompleteDigraph(10), [1, 2]), DigraphColouringAlgorithmByskov);
+10
+gap> ChromaticNumber(Digraph([[4, 8], [6, 10], [9], [2, 3, 9], [],
+> [3], [4], [6], [], [5, 7]]), DigraphColouringAlgorithmByskov);
+3
+gap> ChromaticNumber(DigraphDisjointUnion(CompleteDigraph(1),
+> Digraph([[2], [4], [1, 2], [3]])), DigraphColouringAlgorithmByskov);
+3
+gap> ChromaticNumber(DigraphDisjointUnion(CompleteDigraph(1),
+> Digraph([[2], [4], [1, 2], [3], [1, 2, 3]])), DigraphColouringAlgorithmByskov);
+4
+gap> gr := Digraph([[2, 3, 4], [3], [], []]);
+<immutable digraph with 4 vertices, 4 edges>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+3
+gap> ChromaticNumber(EmptyDigraph(0), DigraphColouringAlgorithmByskov);
+0
+gap> gr := CompleteDigraph(4);;
+gap> gr := DigraphAddVertex(gr);;
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+4
+gap> gr := Digraph([[2, 4, 7, 3], [3, 5, 8, 1], [1, 6, 9, 2],
+> [5, 7, 1, 6], [6, 8, 2, 4], [4, 9, 3, 5], [8, 1, 4, 9], [9, 2, 5, 7],
+> [7, 3, 6, 8]]);;
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+3
+gap> gr := DigraphSymmetricClosure(ChainDigraph(5));
+<immutable symmetric digraph with 5 vertices, 8 edges>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+2
+gap> gr := DigraphFromGraph6String("KmKk~K??G@_@");
+<immutable digraph with 12 vertices, 42 edges>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+4
+gap> gr := CycleDigraph(7);
+<immutable cycle digraph with 7 vertices>
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+3
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+3
+gap> ChromaticNumber(gr, DigraphColouringAlgorithmByskov);
+3
+
```

Comment:
Good tests here!  Did you run your functions on all graphs up to size `n` again, like you did before?  It'd be good to have that extra bit of confidence now you've changed a few things.

Reply:
I kept running the same tests on all graphs up to size 8 as I performed the optimisations, and it still passes them.